roles/barbican/tasks/main.yml

```diff
@@ -0,0 +1,97 @@
+---
+- name: Check if barbican user exists
+  action: shell getent passwd barbican
+  register: barbican_user
+  failed_when: False
+  changed_when: False
+
+- name: barbican user
+  user: name=barbican shell=/bin/false createhome=no
+  when: barbican_user|success
+
+- name: barbican user
+  user: name=barbican comment=barbican shell=/bin/false system=yes home=/nonexistent createhome=no
+  when: barbican_user|failed
+
+- name: /etc/barbican
+  file: dest=/etc/barbican state=directory owner=barbican group=barbican
+
+- name: barbican log dir
+  file: dest=/var/log/barbican state=directory mode=0755 owner=barbican group=adm
+
+- name: Change barbican log dir acl
+  acl: name=/var/log/barbican state=present default=yes etype={{ item.etype }} permissions={{ item.permission }}
+  with_items:
+    - etype: user
+      permission: rw
+    - etype: group
+      permission: r
+    - etype: other
+      permission: r
+      
+- name: create barbican cache dir
+  file: dest=/var/cache/barbican state=directory mode=0700
+        owner=barbican group=barbican
+        
+- name: set uwsgi path (source install)
+  set_fact: uwsgi_path={{ openstack_source.virtualenv_base }}/barbican/bin/uwsgi
+  when: openstack_install_method == 'source'
+
+- name: set uwsgi path (package install ubuntu)
+  set_fact:
+    uwsgi_path: "{{ openstack_package.virtualenv_base }}/barbican/bin/uwsgi"
+  when: openstack_install_method == 'package' and ursula_os == 'ubuntu'
+  
+- name: install barbican uwsgi service (ubuntu)
+  template: src=etc/init/barbican.conf
+            dest=/etc/init/barbican.conf mode=0644
+  when: ursula_os == 'ubuntu'
+
+- name: Distribute Barbican Config Files
+  template: src={{ item }} dest=/etc/barbican mode=0644
+            owner=barbican group=barbican
+  with_fileglob: ../templates/etc/barbican/*
+
+- name: Creates barbican uwsgi and httpd directories
+  file: path={{ item }} state=directory mode=0775
+        owner=barbican group=barbican mode=0775
+  with_items:
+    - /etc/barbican/uwsgi
+    - /run/uwsgi
+    
+- name: configure barbican admin wsgi
+  template: src=etc/uwsgi/barbican-vassals-api.ini
+            dest=/etc/barbican/uwsgi/barbican-main.ini mode=0775
+            owner=barbican group=barbican
+       
+- name: barbican apache vhost (ubuntu)
+  template: src=etc/apache2/sites-available/barbican.conf
+            dest=/etc/apache2/sites-available/barbican.conf
+  when: ursula_os == 'ubuntu'
+  notify:
+    - reload apache
+         
+- name: sync barbican database
```

Comment:
I think we should run this as the barbican user and use the `becomes` ansible parameter. If this is ran before the uwsgi barbican runs, the log file is owned by root and the barbican uwsgi process cannot write to it (`api.log`). I have also noticed that doing `--log-file /var/barbican/barbican-manage.log` is broken and needs to be fixed upstream (probably my fault :p when I put that into barbican-manage).